dask_ml/model_selection/_incremental.py

```diff
@@ -757,7 +859,58 @@ def _get_params(self):
         else:
             return ParameterSampler(self.parameters, self.n_initial_parameters)
 
+    def _stop_on_plateau(self, info, instructions):
+        best_plateau = {}
+        out = {}
+        for k, steps in instructions.items():
+            records = info[k]
+            current_calls = records[-1]["partial_fit_calls"]
+            if self.max_iter and current_calls >= self.max_iter:
+                out[k] = 0
+            elif self.patience and current_calls >= self.patience:
+                plateau = [
+                    h["score"]
+                    for h in records
+                    if current_calls - h["partial_fit_calls"] <= self.patience
+                ]
+                if all(score <= plateau[0] + self.tol for score in plateau):
+                    out[k] = 0
+                    if len(best_plateau) == 0 or plateau[-1] >= max(
+                        best_plateau.values()
+                    ):
+                        best_plateau.update({k: plateau[-1]})
+                else:
+                    out[k] = steps
+
+            else:
+                out[k] = steps
+        return out, best_plateau
+
+    def fit(self, *args, **kwargs):
+        self._best_plateaus = {}
+        self._model_scores = {}
+        self._models_kept = []
+        return super(IncrementalSearchCV, self).fit(*args, **kwargs)
+
     def _additional_calls(self, info):
+        last_scores = {k: v[-1]["score"] for k, v in info.items()}
+        self._model_scores.update(last_scores)
+
+        instructions = self._adapt(info)
+        out, best_plateau = self._stop_on_plateau(info, instructions)
+
+        # Hold onto the models that has performed the best (protects against
+        # scores decreasing or early plateauing with high score
+        models_kept = [
+            k
+            for k, v in self._model_scores.items()
+            if (v >= max(self._model_scores.values()) - 1e-5) and k not in out
+        ]
```

Comment:
What is the 1e-5 number here?

Reply:
Which models do we keep exactly.  After looking at this code for a while I'm still not sure I understand.  

Would it suffice to keep just the best performing model we've seen so far, and also keep track of when we saw it?  In that case then the plateau check might be something like

```python
if score > best_score:
    best_score = score
    best_model = model
    best_time = time  # or partial_fit_calls or whatnot
if time > best_time + patience:
    stop working
```